sgx-tools/gen-qe-target-info.go

```diff
@@ -27,12 +37,35 @@ For example, save the target information file about Quoting Enclave retrieved fr
 			logrus.SetLevel(logrus.DebugLevel)
 		}
 
-		ti, err := intelsgx.GetQeTargetInfo()
+		var ti []byte
+		var err error
+
+		serviceType := context.String("serviceType")
+		AkId, err := GetAttKeyId(serviceType, context.String("AkId"))
 		if err != nil {
-			logrus.Print(err)
 			return err
 		}
 
+		if strings.EqualFold(serviceType, "dcap") {
+			bufSize, _, err := InitQuoteExRequest(AkId, false, uint64(0))
+			if err != nil {
+				return err
+			}
+
+			_, ti, err = InitQuoteExRequest(AkId, true, bufSize)
+			if err != nil {
+				return err
+			}
+		} else if strings.EqualFold(serviceType, "epid") {
+			ti, err = intelsgx.GetQeTargetInfo()
+			if err != nil {
+				logrus.Print(err)
+				return err
+			}
```

Comment:
If your rfc code implements in such a style, it is better to express your intention:

```go
// defined in sgx-tools/xxx.go

func RfcGetQeTargetInfo(type QuoteType) (bytes [], error) {
    if type == ecdsa {

    } else if type == epidUnlinkable || type == epidLinkable {
        // FIXME: currently we don't distinguish  Unlinkable and Linkable EPID-based quote type
        ti, err = intelsgx.GetQeTargetInfo()
    } else {
      // error handing
    }
}

func RfcGetQeTargetInfoByAttestationKeyId(akId bytes[]) (bytes [], error) {
    type, err := RfcGetQuoteTypeByAttestationKeyId(akId)
    if err != nil {
      // error handling
    }

    return RfcGetQeTargetInfo(type)
}
```

Reply:
> ```go
> func RfcGetQeTargetInfoByAttestationKeyId(akId bytes[]) (bytes [], error) {
> ```

`RfcGetQeTargetInfoByAttestationKeyId` means `RfcGetQeQuoteByAttestationKeyId` and other ra functions all must exist. This is not meaningful to implement `ByAttestationKeyId` functions.